src/app/portlets/dot-edit-content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.ts

```diff
@@ -0,0 +1,95 @@
+import { DotDevicesService } from '../../../../api/services/dot-devices/dot-devices.service';
+import { DotLanguagesService } from '../../../../api/services/dot-languages/dot-languages.service';
+import { DotPersonasService } from '../../../../api/services/dot-personas/dot-personas.service';
+import { Persona } from '../../../../shared/models/persona/persona.model';
+import { Observable } from 'rxjs/Observable';
+import { Language } from '../../../../shared/models/language/language.model';
+import { Device } from '../../../../shared/models/device/device.model';
+import { DotViewAsService } from '../../../../api/services/dot-view-as/dot-view-as.service';
+
+@Component({
+    selector: 'dot-edit-content-view-as-toolbar',
+    templateUrl: './dot-edit-content-view-as-toolbar.component.html',
+    styleUrls: ['./dot-edit-content-view-as-toolbar.component.scss']
+})
+export class DotEditContentViewAsToolbarComponent implements OnInit {
+    @Output() changeViewAs = new EventEmitter<DotEditPageViewAs>();
+    @Output() changeDevice = new EventEmitter<Device>();
+
+    languagesOptions: SelectItem[];
+    personasOptions: SelectItem[];
+    devicesOptions: SelectItem[];
+
+    viewAsConfig: DotEditPageViewAs;
+
+    constructor(
+        private dotDevicesService: DotDevicesService,
+        private dotLanguagesService: DotLanguagesService,
+        private dotPersonasService: DotPersonasService,
+        private dotViewAsService: DotViewAsService
+    ) {}
+
+    ngOnInit() {
+        Observable.forkJoin(
+            this.dotPersonasService.get(),
+            this.dotLanguagesService.get(),
+            this.dotDevicesService.get()
+        ).subscribe(response => {
+            this.personasOptions = response[0].map((persona: Persona) => this.getPersonaFieldOption(persona));
+            this.languagesOptions = response[1].map((language: Language) => this.getLanguageFieldOption(language));
+            this.devicesOptions = response[2].map((device: Device) => this.getDeviceFieldOption(device));
+            this.setInitialConfiguration();
+        });
+    }
+
+    /**
+     * Track changes in language and persona.
+     */
+    changeConfiguration() {
+        this.dotViewAsService.selected = this.viewAsConfig;
+        this.changeViewAs.emit(this.dotViewAsService.selected);
+    }
+
+    /**
+     * Track changes in the device.
+     */
+    changeDeviceConfiguration() {
+        this.dotViewAsService.selected = this.viewAsConfig;
+        this.changeDevice.emit(this.viewAsConfig.device);
+    }
+
+
+
+    private setInitialConfiguration() {
+        this.viewAsConfig = this.dotViewAsService.selected ||  {
+            languageId: this.languagesOptions[0].value,
+            personaId: this.personasOptions[0].value,
+            device: this.devicesOptions[0].value
+        };
+        this.changeDevice.emit(this.viewAsConfig.device);
+    }
+
+    private getPersonaFieldOption(persona: Persona): SelectItem {
```

Comment:
This 3 method are pretty much the same... are you sure you can't do just one method for the 3 cases?

Reply:
do they have a common interface, keep in mind the last one does not have an id; not sure if it is intentional or a mistake!

Reply:
It's intentional, still doable.

Reply:
ok in that case just the persona and language could reuse a common method and device will be the same